x-pack/plugin/identity-provider/docs/en/rest-api/idp-saml-init.asciidoc

```diff
@@ -0,0 +1,123 @@
+  "saml_response" : "<?xml version="1.0" encoding="UTF-8"?><saml2p:Response xmlns:saml2p="urn:oasis:names:tc:SAML:2.0:protocol" xmlns:xsd="http://www.w3.org/2001/XMLSchema" Destination="https://sp.some.org/api/security/v1/saml" ID="_845fbfc9f3254162ce1e161c91b07d85311d65cd" IssueInstant="2020-03-19T15:45:00.158Z" ...removed for brevity ... </saml2p:Response>",
+  "saml_status" : "urn:oasis:names:tc:SAML:2.0:status:Success",
+  "error" : null,
+  "service_provider" : {
+    "entity_id" : "https://sp1.kibana.org"
+  }
+}
+--------------------------------------------------------------------
+// TESTRESPONSE[skip:Do not enable identity provider for the docs cluster, at least not yet]
+
+A failed call, in the case of an SP initiated SSO returns a SAML Response as an XML String with its status set to the appropriate error
+code indicating that the authentication request failed and the reason for that failure. A `saml_status` of
+`urn:oasis:names:tc:SAML:2.0:status:Requester` indicates that the error is on the side of the SP or the user, while a `saml_status` of
+`urn:oasis:names:tc:SAML:2.0:status:Responder` indicates that something went wrong in the IDP side. The `error` field contains a short
+human friendly interpretation of the error that is outside the SAML standard and is meant to be communicated to the user, especially
+if the user is not  redirected back the SP with the `saml_response`
+
+[source, console-result]
+--------------------------------------------------------------------
+{
+  "post_url" : "https://sp1.kibana.org/saml/acs",
+  "saml_response" : "?xml version="1.0" encoding="UTF-8"?><saml2p:Response xmlns:saml2p="urn:oasis:names:tc:SAML:2.0:protocol" xmlns:xsd="http://www.w3.org/2001/XMLSchema" Destination="https://sp1.kibana.org/api/saml/acs" ID="_845fbfc9f3254162ce1e161c91b07d85311d65cd" IssueInstant="2020-03-19T15:45:00.158Z" Version="2.0"><saml2:Issuer xmlns:saml2="urn:oasis:names:tc:SAML:2.0:assertion">https://idp.cloud.elastic.co</saml2:Issuer><ds:Signature xmlns:ds="http://www.w3.org/2000/09/xmldsig#">...removed for brevity...</ds:Signature><saml2p:Status><saml2p:StatusCode Value="urn:oasis:names:tc:SAML:2.0:status:Requester"><samlp:StatusCode xmlns:saml2p="urn:oasis:names:tc:SAML:2.0:protocol" Value="urn:oasis:names:tc:SAML:2.0:status:InvalidNameIDPolicy"/></saml2p:StatusCode></saml2p:Status></saml2p:Response>",
+  "saml_status" : "urn:oasis:names:tc:SAML:2.0:status:Requester",
+  "error" : "User [user1] is not permitted to access service [https://sp1.kibana.org]",
+  "service_provider" : {
+    "entity_id" : "https://sp1.kibana.org"
+  }
+}
+--------------------------------------------------------------------
+// TESTRESPONSE[skip:Do not enable identity provider for the docs cluster, at least not yet]
+
+A failed call, in the case of an IDP initiated SSO returns an error with the appropriate HTTP Status code. i.e. if the value of the
+`es-secondary-authorization` is wrong, the IDP will respond with `403`
```

Comment:
From my experimentations, I got 403s when I tried to hit this endpoint with a user that doesn't have access to the resource, and 401s when the `es-secondary-authorization` is wrong.

Reply:
> I got 403s when I tried to hit this endpoint with a user that doesn't have access to the resource, and 401s when the es-secondary-authorization is wrong.

Yes, `403` is a typo in the original description, it should be `401`. `403` is normal/expected when hitting any ES endpoint with a user that is not authorized to do so , thus not sure we will explicitly call this out in this API docs.

> EDIT: and I also wasn't able to to get the 403 to show up when using basic auth..

Not sure what you mean here. Basic auth for `Authorization` header or for `es-secondary-authorization` ? And you were trying to get this for a reason, or because the description was wrong and you expected to get a `403` ?

Reply:
Ah, if 403 is not supposed to be called out in this API, then we can leave it.

In any case, my issue was regarding discrepancy I saw: which is that when I was using basic `es-secondary-authorization`, I didn't get a 403 when using this endpoint to access a resource the user didn't have access to, but I did when I used ApiKey `es-secondary-authorization` auth.

Reply:
Just following up: this was expected behaviour and had nothing to do with ApiKey vs Basic Auth. It was a per-design difference between Sp-initiated vs IdP initiated flow.